### PR TITLE
Replace `assert` with `llvm_unreachable`

### PR DESCRIPTION
**Pull request by texasmichelle**
Add an explicit return of true to ensure all paths return a value. Even though this code path is effectively unreachable, it resolves the build error:

```
In file included from swift/lib/SIL/Verifier/LinearLifetimeChecker.cpp:24:
swift/lib/SIL/Verifier/LinearLifetimeCheckerPrivate.h:212:3: error: non-void function does not return a value in all control paths [-Werror,-Wreturn-type]
  }
  ^
```

### CODE REVIEW

File: lib/SIL/Verifier/LinearLifetimeCheckerPrivate.h [hunk 1]
```diff
@@ -209,6 +209,7 @@ class LLVM_LIBRARY_VISIBILITY LinearLifetimeChecker::ErrorBuilder {
 
     llvm::errs() << "Found ownership error?!\n";
     assert(0 && "triggering standard assertion failure routine");
+    return true;
```
Comment (DougGregor) [inline lib/SIL/Verifier/LinearLifetimeCheckerPrivate.h:212]: `llvm_unreachable("message")` is the canonical way to write `assert(0 && "message");`, and should avoid this failure

Comment (texasmichelle) [reply lib/SIL/Verifier/LinearLifetimeCheckerPrivate.h:212]: Will `llvm_unreachable()` avoid the PrettyStackTrack swallowing?
https://github.com/apple/swift/pull/32500

Comment (gottesmm) [reply lib/SIL/Verifier/LinearLifetimeCheckerPrivate.h:212]: Yea. It will be fine. I don't remember why I did it this way. I was pretty working quickly.

Comment (gottesmm) [reply lib/SIL/Verifier/LinearLifetimeCheckerPrivate.h:212]: The problem in that other one was llvm::report_fatal_error.

**Comment by gottesmm** [comment]
Can you change this to llvm_unreachable and then LGTM?

**Review by gottesmm — APPROVED** [review]

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Pull request merged**